xonsh/ptk_shell/shell.py

```diff
@@ -76,6 +78,23 @@ def tokenize_ansi(tokens):
     return ansi_tokens
 
 
+def remove_ansi_osc(prompt):
+    """Removes the ANSI OSC escape codes - ``prompt_toolkit`` does not support them.
+    Some terminal emulators - like iTerm2 - uses them for various things.
+
+    See: https://www.iterm2.com/documentation-escape-codes.html
+    """
+
+    osc_tokens = ANSI_OSC_PATTERN.findall(prompt)
+    prompt = ANSI_OSC_PATTERN.sub("", prompt)
+
+    return prompt, osc_tokens
+
+
+def write_ansi_osc(ansi_osc_code):
```

Comment:
Please add a docstring to public functions

Reply:
Since it's only a single `print` now, I removed the whole function.

Reply:
Cool, even better!